shaders/src/shadowing.fs

```diff
@@ -327,3 +350,15 @@ float shadow(const lowp sampler2DArrayShadow shadowMap, const uint layer, const
     return ShadowSample_PCF_High(shadowMap, layer, size, shadowPosition);
 #endif
 }
+
+float shadowVsm(const lowp sampler2DArray shadowMap, const uint layer, const vec3 shadowPosition,
```

Comment:
You should double check the precision qualifiers here. shadowMap shouldn't be lowp but highp (for 32F depth maps, or mediump otherwise). The shadow position and fragDepth require highp as well.